be/src/vec/exec/vaggregation_node.cpp

```diff
@@ -499,11 +508,14 @@ Status AggregationNode::close(RuntimeState* state) {
     VExpr::close(_probe_expr_ctxs, state);
     if (_executor.close) _executor.close();
 
-    std::visit(
-            [&](auto&& agg_method) {
-                COUNTER_SET(_hash_table_size_counter, int64_t(agg_method.data.size()));
-            },
-            _agg_data._aggregated_method_variant);
+    /// _hash_table_size_counter may be null if prepare failed.
+    if (_hash_table_size_counter) {
```

Comment:
If prepare failed, `close` should be never called.

Reply:
Actually in function `Status FragmentMgr::exec_plan_fragment(const TExecPlanFragmentParams& params, FinishCallback cb)` when prepare was failed, the destructor of `PlanFragmentExecutor` will call `close`.